benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -694,28 +700,34 @@ void GraphicsBenchmarkApp::ProcessKnobs()
 {
     // Detect if any knob value has been changed
     // Note: DigestUpdate should be called only once per frame (DigestUpdate unflags the internal knob variable)!
-    const bool allTexturesTo1x1KnobChanged = pAllTexturesTo1x1->DigestUpdate();
-    const bool alphaBlendKnobChanged       = pAlphaBlend->DigestUpdate();
-    const bool depthTestWriteKnobChanged   = pDepthTestWrite->DigestUpdate();
-    const bool enableSpheresKnobChanged    = pEnableSpheres->DigestUpdate();
+    const bool allTexturesTo1x1KnobChanged   = pAllTexturesTo1x1->DigestUpdate();
+    const bool alphaBlendKnobChanged         = pAlphaBlend->DigestUpdate();
+    const bool depthTestWriteKnobChanged     = pDepthTestWrite->DigestUpdate();
+    const bool enableSpheresKnobChanged      = pEnableSpheres->DigestUpdate();
+    const bool initializedSpheresKnobChanged = pInitializedSpheres->DigestUpdate();
 
     const bool enableSpheres = pEnableSpheres->GetValue();
     // If the LOD is empty, assuming we skipped the setup for all sphere resources at start
     // So need to do the intial setup for all resources here
     const bool spheresAreSetUp               = !mSphereLODs.empty();
     const bool updateSphereDescriptors       = spheresAreSetUp && allTexturesTo1x1KnobChanged;
-    const bool setupSphereResourcesAndMeshes = (!spheresAreSetUp) && enableSpheres;
+    const bool setupSphereResourcesAndMeshes = (!spheresAreSetUp || initializedSpheresKnobChanged) && enableSpheres;
```

Comment:
Sphere resources are not cleaned up. Right now, we re-create new descriptors while keeping the old ones, exhausting the pool. This needs to be fixed before this knob can be used.

The other issue is we initialize spheres twice by default:
 - once because the pEnableSpheres bit is set by default.
 - then here, because the initializedSpheresKnob by default is marked as changed.

Reply:
Updated.